crates/rpc/rpc-eth-api/src/cache/db.rs

```diff
@@ -0,0 +1,169 @@
+//! Helper types to workaround 'higher-ranked lifetime error'
+//! <https://github.com/rust-lang/rust/issues/100013> in default implementation of
+//! [`Call`](crate::servers::Call).
+
+use reth_primitives::{B256, U256};
+use reth_provider::StateProvider;
+use reth_revm::{database::StateProviderDatabase, db::CacheDB, DatabaseRef};
+use revm::Database;
+
+/// Helper alias type for the state's [`CacheDB`]
+pub type StateCacheDb<'a> = CacheDB<StateProviderDatabase<StateProviderTraitObjWrapper<'a>>>;
+
+/// Hack to get around 'higher-ranked lifetime error', see
+/// <https://github.com/rust-lang/rust/issues/100013>
+#[allow(missing_debug_implementations)]
+pub struct StateProviderTraitObjWrapper<'a>(pub &'a dyn StateProvider);
```

Comment:
I kinda have an idea why this is happening, 
there's probably a way around this, but we can explore this separately